cmd/build.go

```diff
@@ -5,94 +5,172 @@
+	ignore            []string
+	debug             bool
 }
 
-func build(args []string) error {
+func newBuildParams() buildParams {
+	var buildParams buildParams
+	buildParams.target = util.NewEnumFlag(compile.TargetRego, []string{compile.TargetRego, compile.TargetWasm})
+	return buildParams
+}
 
-	ctx := context.Background()
+func init() {
 
-	f := loaderFilter{
-		Ignore: buildParams.ignore,
-	}
+	buildParams := newBuildParams()
 
-	regoArgs := []func(*rego.Rego){
-		rego.Query(args[0]),
-	}
+	var buildCommand = &cobra.Command{
+		Use:   "build <path> [<path> [...]]",
+		Short: "Build an OPA bundle",
+		Long: `Build an OPA bundle.
+
+The 'build' command packages OPA policy and data files into bundles. Bundles are
+gzipped tarballs containing policies and data. Paths referring to directories are
+loaded recursively.
+
+	$ ls
+	example.rego
+
+	$ opa build -b .
+
+You can load bundles into OPA on the command-line:
+
+	$ ls
+	bundle.tar.gz example.rego
+
+	$ opa run bundle.tar.gz
+
+You can also configure OPA to download bundles from remote HTTP endpoints:
+
+	$ opa run --server \
+		--set bundles.example.resource=bundle.tar.gz \
+		--set services.example.url=http://localhost:8080
+
+Inside another terminal in the same directory, serve the bundle via HTTP:
+
+	$ python3 -m http.server --bind localhost 8080
+
+For more information on bundles see https://www.openpolicyagent.org/docs/latest/management.
+
+When -b is specified the 'build' command assumes paths refer to existing bundle files
+or directories following the bundle structure. If multiple bundles are provided, their
+contents are merged. If there are any merge conflicts (e.g., due to conflicting bundle
+roots), the command fails.
+
+The -O flag controls the optimization level. By default, optimization is disabled (-O=0).
+When optimization is enabled the 'build' command generates a bundle that is semantically
+equivalent to the input files however the structure of the files in the bundle may have
+been changed by rewriting, inlining, pruning, etc. Higher optimization levels may result
+in longer build times.
+
+The 'build' command supports targets (specifie by -t):
 
-	if buildParams.dataPaths.isFlagSet() {
-		regoArgs = append(regoArgs, rego.Load(buildParams.dataPaths.v, f.Apply))
+    rego    The default target emits a bundle containing a set of policy and data files
+            that are semantically equivalent to the input files. If optimizations are
+            disabled the output may simply contain a copy of the input policy and data
+            files. If optimization is enabled at least one entrypoint (-e) must be supplied.
+
+    wasm    The wasm target emits a bundle containing a WebAssembly module compiled from
+            the input files. The bundle may contain the original policy or data files.
+            The wasm target requires exactly one entrypoint (-e) be supplied.
+
+The -e flag tells the 'build' command that a specific document may be queried by the
```

Comment:
I read this line a couple times before understanding it.  Maybe....

The -e flag tells the 'build' command which documents will be queried by the software asking for policy decisions, so that it can focus optimization efforts and ensure that document is not eliminated by the optimizer.

Reply:
Fixed.